src/interpolator.ts

```diff
@@ -117,8 +117,8 @@ export class Interpolator {
     const clamp0 = ctrl0.signed ? sclamp0 : uclamp0;
 
     const alpha1 = result1 & 0xff;
-    const ublend1 = u32(this.base0) + (((alpha1 * (u32(this.base1) - u32(this.base0))) / 256) | 0);
-    const sblend1 = s32(this.base0) + (((alpha1 * (s32(this.base1) - s32(this.base0))) / 256) | 0);
+    const ublend1 = u32(this.base0) + (Math.floor((alpha1 * (u32(this.base1) - u32(this.base0))) / 256) | 0);
```

Comment:
I can't shift right by 8 bits because JS will cast the value to a 32-bit int before doing the bit operations. The multiplication result can be an up to 40 bit intermediate value, and we would lose the top 8 bits if we just did `>>> 8`

Reply:
Got it, then I'll merge it as-is